### PR TITLE
NextMillennium Bid Adapter: Add referrer and imp to bid request

### PR DESCRIPTION
**Pull request by JacobKlein26**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

File: modules/nextMillenniumBidAdapter.js [hunk 1]
```diff
@@ -24,12 +24,18 @@ export const spec = {
 
     _each(validBidRequests, function(bid) {
       window.nmmRefreshCounts[bid.adUnitCode] = window.nmmRefreshCounts[bid.adUnitCode] || 0;
+      const id = getPlacementId(bid)
+
+      if (bid.sizes && !Array.isArray(bid.sizes[0])) bid.sizes = [bid.sizes]
+      if (!bid.ortb2) bid.ortb2 = {}
+      if (!bid.ortb2.device) bid.ortb2.device = {}
+      bid.ortb2.device.refferer = getWindowTop().document && getWindowTop().document.referrer
```
Comment (patmmccann) [inline modules/nextMillenniumBidAdapter.js:32]: looks like a typo in 'refferer'

**Review by patmmccann — CHANGES_REQUESTED** [review]

File: modules/nextMillenniumBidAdapter.js [hunk 2]
```diff
@@ -24,12 +24,18 @@ export const spec = {
 
     _each(validBidRequests, function(bid) {
       window.nmmRefreshCounts[bid.adUnitCode] = window.nmmRefreshCounts[bid.adUnitCode] || 0;
+      const id = getPlacementId(bid)
+
+      if (bid.sizes && !Array.isArray(bid.sizes[0])) bid.sizes = [bid.sizes]
+      if (!bid.ortb2) bid.ortb2 = {}
+      if (!bid.ortb2.device) bid.ortb2.device = {}
+      bid.ortb2.device.referrer = getWindowTop().document && getWindowTop().document.referrer
```
Comment (patmmccann) [inline modules/nextMillenniumBidAdapter.js:32]: why not use getRefererInfo().page ?

**Review by patmmccann — CHANGES_REQUESTED** [review]

File: modules/nextMillenniumBidAdapter.js (hunk 2 above)
Comment (JacobKlein26) [reply modules/nextMillenniumBidAdapter.js]: I updated it, 
I just didn't know about this...

Thanks!

**Comment by JacobKlein26** [comment]
@patmmccann 
I believe I addresses all requested changes, 
Please let me know if I missed anything 

Thank you

**Review by patmmccann — APPROVED** [review]

File: modules/nextMillenniumBidAdapter.js [hunk 3]
```diff
@@ -24,12 +25,18 @@ export const spec = {
 
     _each(validBidRequests, function(bid) {
       window.nmmRefreshCounts[bid.adUnitCode] = window.nmmRefreshCounts[bid.adUnitCode] || 0;
+      const id = getPlacementId(bid)
+
+      if (bid.sizes && !Array.isArray(bid.sizes[0])) bid.sizes = [bid.sizes]
+      if (!bid.ortb2) bid.ortb2 = {}
+      if (!bid.ortb2.device) bid.ortb2.device = {}
+      bid.ortb2.device.referrer = (getRefererInfo && getRefererInfo().ref) || ''
```
Comment (JacobKlein26) [inline modules/nextMillenniumBidAdapter.js:33]: @patmmccann, 
Thank you so much for reviewing and approving this PR,
This is currently live, but is however creating an unexpected error, which I can't figure out why.
I know it's not your job, but I would greatly appreciate it if you can maybe help me out here for a quick second, 

I'll attach a screenshot of the issue, but in short the bid.ortb2 object is empty, and my fallback line for such a case (32) seems to not work (as it's still an empty object in line 33)

image

If you want to see a case where this is happening you can see it on [this](https://tettybetty.com/feel-good-surfing-art-by-jack-soren/?pb-debug=true) page

**Pull request merged**